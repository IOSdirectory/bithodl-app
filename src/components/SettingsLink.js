import React, { Component } from 'react';
import { StyleSheet, View, TouchableHighlight } from 'react-native';
import PropTypes from 'prop-types';
import Icon from 'react-native-vector-icons/Ionicons';
import StyledText from './StyledText';

const styles = StyleSheet.create({
  container: {
    paddingTop: 12,
    paddingBottom: 12,
    paddingRight: 15,
    marginLeft: 15,
    borderBottomColor: '#DDDDDF',
    borderBottomWidth: StyleSheet.hairlineWidth
  },
  label: {
    fontWeight: '600'
  },
  value: {
    position: 'absolute',
    top: 12,
    right: 30,
    color: '#8F8E94',
    textAlign: 'right',
    width: 230,
    backgroundColor: 'transparent'
  },
  chevron: {
    position: 'absolute',
    top: 12,
    right: 15,
    fontSize: 20,
    color: '#EBEBEB'
  }
});

export default class SettingsLink extends Component {
  render() {
    const isLastItem = this.props.isLastItem;

    const containerStyles = [
      styles.container,
      isLastItem ? { borderBottomWidth: 0 } : undefined
    ];

    return (
      <TouchableHighlight onPress={this.props.onPress} underlayColor='#FAFAFA'>
        <View style={containerStyles}>
          <StyledText style={styles.label}>{this.props.name}</StyledText>
          <StyledText style={styles.value} numberOfLines={1}>{this.props.value}</StyledText>
          <Icon name='ios-arrow-forward' style={styles.chevron} />
        </View>
      </TouchableHighlight>
    );
  }
}

SettingsLink.propTypes = {
  name: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  value: PropTypes.string,
  isLastItem: PropTypes.bool
};
