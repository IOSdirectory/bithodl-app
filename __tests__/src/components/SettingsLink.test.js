import React from 'react';
import renderer from 'react-test-renderer';
import SettingsLink from '../../../src/components/SettingsLink';

describe('SettingsLink', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SettingsLink
        name='72b47a5f-db79-4431-a2ca-612d97f547cc'
        value='e87b6472-bbd7-4a6d-833a-0c5030189213'
        onPress={() => {}}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly as the last item', () => {
    const tree = renderer.create(
      <SettingsLink
        name='d8d6a18e-0553-44fa-9f68-158b7364bc5e'
        value='b40740ec-ccb1-41c6-8fa5-72147f467556'
        onPress={() => {}}
        isLastItem={true}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
