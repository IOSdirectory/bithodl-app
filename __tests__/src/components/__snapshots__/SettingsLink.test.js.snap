// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsLink renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#DDDDDF",
          "borderBottomWidth": 0.5,
          "marginLeft": 15,
          "paddingBottom": 12,
          "paddingRight": 15,
          "paddingTop": 12,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Object {
            "fontWeight": "600",
          },
        ]
      }
    >
      72b47a5f-db79-4431-a2ca-612d97f547cc
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Object {
            "backgroundColor": "transparent",
            "color": "#8F8E94",
            "position": "absolute",
            "right": 30,
            "textAlign": "right",
            "top": 12,
            "width": 230,
          },
        ]
      }
    >
      e87b6472-bbd7-4a6d-833a-0c5030189213
    </Text>
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#EBEBEB",
            "fontSize": 20,
            "position": "absolute",
            "right": 15,
            "top": 12,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`SettingsLink renders correctly as the last item 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#DDDDDF",
          "borderBottomWidth": 0.5,
          "marginLeft": 15,
          "paddingBottom": 12,
          "paddingRight": 15,
          "paddingTop": 12,
        },
        Object {
          "borderBottomWidth": 0,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Object {
            "fontWeight": "600",
          },
        ]
      }
    >
      d8d6a18e-0553-44fa-9f68-158b7364bc5e
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Object {
            "backgroundColor": "transparent",
            "color": "#8F8E94",
            "position": "absolute",
            "right": 30,
            "textAlign": "right",
            "top": 12,
            "width": 230,
          },
        ]
      }
    >
      b40740ec-ccb1-41c6-8fa5-72147f467556
    </Text>
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#EBEBEB",
            "fontSize": 20,
            "position": "absolute",
            "right": 15,
            "top": 12,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
